tartufo/scanner.py

```diff
@@ -32,19 +34,108 @@
 PatternDict = Dict[str, Union[str, Pattern]]
 
 
-class Bcolors:
-    HEADER = "\033[95m"
-    OKBLUE = "\033[94m"
-    OKGREEN = "\033[92m"
-    WARNING = "\033[93m"
-    FAIL = "\033[91m"
-    ENDC = "\033[0m"
-    BOLD = "\033[1m"
-    UNDERLINE = "\033[4m"
-
-    def __init__(self, name):
-        # type: (str) -> None
-        self.name = name
+class IssueType(enum.Enum):
+    Entropy = "High Entropy"
+    RegEx = "Regular Expression Match"
+
+
+class Issue:
+    """Represents an issue found while scanning the code."""
+
+    OUTPUT_SEPARATOR = "~~~~~~~~~~~~~~~~~~~~~"  # type: str
+    DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S"  # type: str
+
+    issue_type = None  # type: Optional[IssueType]
```

Comment:
What is the advantage of this vs
```IssueType issue_type = None``` for the declaration?

Reply:
Well, the advantage of this way is that it's valid Python syntax. :wink: 
The proper inline annotation in Python would be:
```python
issue_type: IssueType = None
```
As described in [PEP 526](https://www.python.org/dev/peps/pep-0526/#abstract). But, this particular syntax is only available in Python 3.6 and above, meaning we would have to drop Python 3.5 support in order to use this syntax. A move like this is not entirely unprecedented, for example in the [black](https://github.com/psf/black) autoformatter. But given that Python 3.5 is still [officially supported until 2020-09-13](https://devguide.python.org/#branchstatus) I'm a little bit hesitant to just go ahead and drop that support now. But of course, that's not off the table. :)